json-schemas/src/client-events-connections.json

```diff
@@ -76,6 +76,25 @@
       "type": "integer",
       "description": "The duration the connection was open for in milliseconds."
     },
+    "credentials": {
+      "type": "object",
+      "description": "The connection's token credentials.",
+      "properties": {
+        "hasUserClaims": {
+          "type": "boolean",
+          "description": "Does this token contain user claims."
+        },
+        "keyName": {
+          "type": "string",
+          "description": "The token's key name."
+        },
+        "type": {
+          "type": "string",
+          "description": "The credential type: either 'key', 'ablyToken', 'ablyJWT', or 'ablyJWE'."
+        }
+      },
+      "required": ["hasUserClaims", "type"]
```

Comment:
Why isn't `keyName` required?

Reply:
According to the TS definition `this.authParams.getKeyUsername()` can return a string or null, however looking at the implementation it seems that this is not correct: a string is returned in all cases. I'll fix that.